src/submission/validation-clinical/specimen.ts

```diff
@@ -137,3 +139,75 @@ const getDataFromDonorRecordOrDonor = (
 
   return { donorVitalStatus, donorSurvivalTime };
 };
+
+const checkRequiredFields = (
+  specimen: DeepReadonly<Specimen>,
+  specimenRecord: DeepReadonly<SubmittedClinicalRecord>,
+  errors: SubmissionValidationError[],
+) => {
+  const requiredFieldsForTumour: Array<keyof typeof SpecimenFieldsEnum> = [
+    'pathological_tumour_staging_system',
+    'pathological_stage_group',
+    'tumour_grading_system',
+    'tumour_grade',
+    'percent_tumour_cells',
+    'percent_proliferating_cells',
+    'percent_stromal_cells',
+    'percent_necrosis',
+  ];
+
+  const optionalFieldsForTumour: Array<keyof typeof SpecimenFieldsEnum> = [
+    'pathological_t_category',
+    'pathological_n_category',
+    'pathological_m_category',
+  ];
+
+  const isValueMissing = (value: string | number | boolean | undefined) =>
+    isAbsent(value) || (typeof value === 'string' && isEmptyString(value));
+
+  if (specimen.tumourNormalDesignation === 'Tumour') {
+    const missingRequiredFields = requiredFieldsForTumour.filter(field =>
+      isValueMissing(specimenRecord[field]),
+    );
+    missingRequiredFields.forEach(field => {
+      errors.push(
+        utils.buildSubmissionError(
+          specimenRecord,
+          DataValidationErrors.MISSING_VARIABLE_REQUIREMENT,
+          SpecimenFieldsEnum[field],
+          {
+            submitter_specimen_id: specimenRecord['submitter_specimen_id'],
+            referenceSchema: ClinicalEntitySchemaNames['REGISTRATION'],
```

Comment:
use the entity type constants

Reply:
oh that's the enum key, I thought it was the entity name.. you can call it ClinicalEntitySchemaNames.REGISTRATION instead of using the string, so it's easier to rename if needed